Extension/src/LanguageServer/client.ts

```diff
@@ -1980,123 +1999,128 @@ export class DefaultClient implements Client {
+            if (configs && configs.length > 0) {
+                this.sendCustomConfigurations(configs, provider.version);
+            }
+            onFinished();
+        } catch (err) {
+            if (requestFile) {
                 onFinished();
-            } catch (err) {
-                if (requestFile) {
-                    onFinished();
+                return;
+            }
+            const settings: CppSettings = new CppSettings(this.RootUri);
+            if (settings.configurationWarnings === true && !this.isExternalHeader(docUri) && !vscode.debug.activeDebugSession) {
+                const dismiss: string = localize("dismiss.button", "Dismiss");
+                const disable: string = localize("diable.warnings.button", "Disable Warnings");
+                const configName: string | undefined = this.configuration.CurrentConfiguration?.name;
+                if (!configName) {
                     return;
                 }
-                const settings: CppSettings = new CppSettings(this.RootUri);
-                if (settings.configurationWarnings === true && !this.isExternalHeader(docUri) && !vscode.debug.activeDebugSession) {
-                    const dismiss: string = localize("dismiss.button", "Dismiss");
-                    const disable: string = localize("diable.warnings.button", "Disable Warnings");
-                    const configName: string | undefined = this.configuration.CurrentConfiguration?.name;
-                    if (!configName) {
-                        return;
-                    }
-                    let message: string = localize("unable.to.provide.configuration",
-                        "{0} is unable to provide IntelliSense configuration information for '{1}'. Settings from the '{2}' configuration will be used instead.",
-                        providerName, docUri.fsPath, configName);
-                    if (err) {
-                        message += ` (${err})`;
-                    }
+                let message: string = localize("unable.to.provide.configuration",
+                    "{0} is unable to provide IntelliSense configuration information for '{1}'. Settings from the '{2}' configuration will be used instead.",
+                    provider.name, docUri.fsPath, configName);
+                if (err) {
+                    message += ` (${err})`;
+                }
 
-                    if (await vscode.window.showInformationMessage(message, dismiss, disable) === disable) {
-                        settings.toggleSetting("configurationWarnings", "enabled", "disabled");
-                    }
+                if (await vscode.window.showInformationMessage(message, dismiss, disable) === disable) {
+                    settings.toggleSetting("configurationWarnings", "enabled", "disabled");
                 }
             }
-        });
+        } finally {
+            DefaultClient.isStarted.resolve();
```

Comment:
There is an early return before this try/finally on line 2021.  What happens if this `resolve` doesn't happen?

Reply:
OOh. good catch.

Reply:
fixed